### PR TITLE
Refactor IPC tests and enable them for OS provider

### PR DESCRIPTION
**Pull request by vinser52**
### Description
Minor refactoring of IPC. Enabled IPC tests for the OS provider.

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly
- [x] New tests added, especially if they will fail without my changes

### CODE REVIEW

**Comment by lukaszstolarczuk** [comment]
> @bratpiorka As I understand, CI fails because the `mmap` is failed inside the `os_open_ipc_handle` function. According to the log, it failed not immediately but after some iterations. I guess it is because the IPC cache is not implemented yet. But locally tests are passed. Is there a way to change CI runner parameters to allow more virtual memory to be used by the process?

The tests that failed are running on the VMs delivered to us, not defined by us. If you can change it in the system - just update our CI workflow; if you can't change it we have to update the test, I guess.

Perhaps, we could also enable this specific test only in a selected workflow - e.g. multi-numa (which has our own machine used for CI).

**Comment by vinser52** [comment]
> > @bratpiorka As I understand, CI fails because the `mmap` is failed inside the `os_open_ipc_handle` function. According to the log, it failed not immediately but after some iterations. I guess it is because the IPC cache is not implemented yet. But locally tests are passed. Is there a way to change CI runner parameters to allow more virtual memory to be used by the process?
> 
> The tests that failed are running on the VMs delivered to us, not defined by us. If you can change it in the system - just update our CI workflow; if you can't change it we have to update the test, I guess.
> 
> Perhaps, we could also enable this specific test only in a selected workflow - e.g. multi-numa (which has our own machine used for CI).

Ok, thank you, @lukaszstolarczuk. 
For now I enabled only OS provider with Disjoint pool combination. We can enable tests with a scalable pool once IPC caching is done.

File: src/ipc.c [hunk 1]
```diff
@@ -51,6 +51,11 @@ umf_result_t umfPoolGetIPCHandleSize(umf_memory_pool_handle_t hPool,
 
 umf_result_t umfGetIPCHandle(const void *ptr, umf_ipc_handle_t *umfIPCHandle,
                              size_t *size) {
+    if (ptr == NULL || umfIPCHandle == NULL || size == NULL) {
+        LOG_ERR("invalid arguments.");
```
Comment (ldorau) [inline src/ipc.c:55]: `"invalid argument"`

**Review by ldorau — APPROVED** [review]

File: src/ipc.c (hunk 1 above)
Comment (vinser52) [reply src/ipc.c]: Done

**Review by ldorau — APPROVED** [review]

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**